app/models/user.rb

```diff
@@ -1,11 +1,20 @@
 class User < ActiveRecord::Base
   has_secure_password validations: false
-  has_many :reviews
+  has_many :reviews, -> { order("created_at DESC") }
   has_many :queue_items, -> { order(:list_order) }
+  has_many :following_relationships, class_name: 'Relationship', foreign_key: :follower_id
+  has_many :leading_relationships, class_name: 'Relationship', foreign_key: :leader_id
 
   validates_presence_of :email, :password, :full_name
   validates_uniqueness_of :email
 
+  def can_follow?(another_user)
+    self != another_user && !follows?(another_user)
+  end
+
+  def follows?(another_user)
+    following_relationships.map(&:leader).include?(another_user)
```

Comment:
@brandoncc This is an interesting scenario. On the one hand, I could create an array of all the leader_ids in memory and then scan through that for another user id. Or I could only find the relevant data, though this does involve instantiating a more complex object.  
Though, since a user can only follow someone once, both find_by and where would at one return 1 object. With that in mind, I think the latter two options may be a bit faster.